src/cpp/shared/combat/animation.cpp

```diff
@@ -1,149 +1,79 @@
 #include <string.h>
 #include "combat/animation.h"
 
+static void AllocAnimation(SMonFrameInfo *frm, int animType, char *animInfo) {
+  memcpy(&frm->animationFrameToImgIdx[ANIMATION_TYPE_WALKING][frm->animationLengths[ANIMATION_TYPE_WALKING]],
+          frm->animationFrameToImgIdx[animType],
+          frm->animationLengths[animType]);
+  memcpy(&frm->moveAnimInfoIgnored[frm->animationLengths[ANIMATION_TYPE_WALKING]],
+          animInfo,
+          frm->animationLengths[animType]);
+  frm->animationLengths[ANIMATION_TYPE_WALKING] += frm->animationLengths[animType];
+}
+
+static void AllocAnimation2(SMonFrameInfo *frm, int animType, char *animInfo) {
+  memcpy(&frm->animationFrameToImgIdx[ANIMATION_TYPE_WALKING][frm->animationLengths[ANIMATION_TYPE_WALKING]],
+          frm->animationFrameToImgIdx,
```

Comment:
I think this is equivalent to AllocAnimation with animType == 0, i.e.: animation ANIMATION_TYPE_START_MOVE .